src/types/bytes.rs

```diff
@@ -121,6 +124,18 @@ impl<I: SliceIndex<[u8]>> Index<I> for PyBytes {
     }
 }
 
+/// Special-purpose trait impl to efficiently handle both [`PyBytes`] and [`PyByteArray`]
```

Comment:
This should also describe the performance characteristics:

```suggestion
/// Special-purpose trait impl to efficiently handle both [`PyBytes`] and [`PyByteArray`]
///
/// If the source object is a `bytes` object, the `Cow` will be borrowed and pointing into the source object, and no copying or heap allocations will happen. If it is a `bytearray`, its contents will be copied to an owned `Cow`.
```